ItemsListApp/test/ItemsListApp.Api.UnitTests/Helpers/ItemTestCaseBuilder.cs

```diff
@@ -5,47 +5,56 @@
 
 namespace ItemsListApp.Api.UnitTests.Helpers
 {
-    internal class ItemTestCaseBuilder
+    internal abstract class ItemTestCaseBuilder
```

Comment:
This is really great. I love how you extended the pattern for `PUT` tests and once we restructured it a bit, it really makes a good sense and fits greatly to the solution.

I consider following to be too much work and quite above the task, however, if you implemented such builder in a production code, you would go extra step to prevent unintentional misuse of the class, you would:
* Create an internal `ItemTestCaseBuilderFactory` which would accommodate the two implementations of yours as private classes and expose only 2 methods (`Use[Post|Put]Builder`)
* Create an interface that would allow user to either call `ToBuildNullItemTestCase` or decide to `ToCreateInvalidItem` - this interface would be returned by both factory methods
* Create an interface returned by the later method above that would allow user to call any of invalidation methods (preferably renamed to start with `With` prefix) and each returning following interface
* Create an interface that would extend the interface above, but include `Build` method on top of the others
* Let the abstract class implement all the interfaces

This would lead to a code that would read something like this:
````
ItemTestCaseBuilderFactory
  .UsePostBuilder()
  .ToCreateInvalidItem()
  .WithInvalidText(…)
  .WithInvalidCreationDate(…)
  .WithInvalidIdentifier(…)
  .Build();
````
One would actually only need to trivially implement (`=> this;`) one extra method in the abstract class (and rename existing, of course) and, as a bonus, would not allow user (a fellow tests writer such as your future self) to apply methods in an order (or in this case "a way") that would lead to invalid test case. But again, tests are not a very good place to fully use this pattern. Especially since you usually call but one method…